ui/app/src/components/shared/ResourceCard.tsx

```diff
@@ -40,122 +40,116 @@ export const ResourceCard: React.FunctionComponent<ResourceCardProps> = (props:
       || (props.resource.azureStatus?.powerState && props.resource.azureStatus.powerState !== VMPowerStates.Running)
   }
 
-  const resourceCosts = costsCtx?.costs?.find((resourceCost) => {
-    return resourceCost.id === props.resource.id;
-  });
-
-  let resourceUrl = ""
-  switch(props.resource.resourceType) {
-    case ResourceType.Workspace:
-    case ResourceType.WorkspaceService:
-    case ResourceType.UserResource:
-      resourceUrl = props.resource.resourcePath;
-      break;
-    case ResourceType.SharedService: // shared services are accessed from the root and the workspace, have to handle the URL differently
-      resourceUrl = workspaceCtx.workspace ? props.resource.id : props.resource.resourcePath;
-      break;
+  const goToResource = useCallback(() => {
+    let resourceUrl = '';
+    switch(props.resource.resourceType) {
+      case ResourceType.Workspace:
+      case ResourceType.WorkspaceService:
+      case ResourceType.UserResource:
+        resourceUrl = props.resource.resourcePath;
+        break;
+      case ResourceType.SharedService: // shared services are accessed from the root and the workspace, have to handle the URL differently
+        resourceUrl = workspaceCtx.workspace ? props.resource.id : props.resource.resourcePath;
+        break;
+    }
+
+    props.selectResource && props.selectResource(props.resource);
+    navigate(resourceUrl);
+  }, [navigate, props, workspaceCtx.workspace]);
+
+  const resourceStatus = latestUpdate.operation?.status
+    ? latestUpdate.operation.status
+    : props.resource.deploymentStatus
+
+  // Decide what to show as the top-right header badge
+  let headerBadge = <></>;
+  if (
+    latestUpdate.componentAction !== ComponentAction.Lock &&
+    props.resource.azureStatus?.powerState &&
+    successStates.includes(resourceStatus) &&
+    props.resource.isEnabled
+  ) {
+    headerBadge = <PowerStateBadge state={props.resource.azureStatus.powerState} />
+  } else {
+    headerBadge = <StatusBadge resource={props.resource} status={resourceStatus} />
   }
 
+  const authNotProvisioned = props.resource.resourceType === ResourceType.Workspace && !props.resource.properties.scope_id;
```

Comment:
i'm not seeing this tooltip on non-auth provisioned cards? also - can we prevent the animation / cursor changing to a pointer if the auth hasn't been provisioned? currently it's giving the signals that it's clickable, but nothing happens when I click

Reply:
Have you got the latest? Added a fix for that not long ago

Reply:
You should see this: 
![image](https://user-images.githubusercontent.com/25235950/200811798-f77b4c1d-4aba-4d7e-b994-6f3c0c325dce.png)

Reply:
yep, see it now, how odd as I pulled this morning. Like it!